ChessBoard.UnitTests/ChessBoardTests.cs

```diff
@@ -0,0 +1,82 @@
+using System.Text;
+using Microsoft.Extensions.Logging;
+using Moq;
+
+namespace ChessBoardApp.ChessBoard.Tests
+{
+    class TestLogger<T> : ILogger<T>
+    {
+        public List<string> LoggedMessages { get; } = new();
+
+        public IDisposable BeginScope<TState>(TState state) => null;
+
+        public bool IsEnabled(LogLevel logLevel) => true;
+
+        public void Log<TState>(
+            LogLevel logLevel,
+            EventId eventId,
+            TState state,
+            Exception exception,
+            Func<TState, Exception, string> formatter
+        )
+        {
+            string logMessage = formatter(state, exception);
+            LoggedMessages.Add(logMessage);
+        }
+    }
+
+    public class ChessBoardGeneratorTests
+    {
+        [Fact]
+        public void GenerateRectangleBoard_ShouldLogCorrectChessBoard()
+        {
+            // Arrange
+            var logger = new TestLogger<ChessBoardGenerator>();
+            var chessBoardGenerator = new ChessBoardGenerator(logger);
+            int width = 4;
+            int length = 6;
+
+            // Act
+            chessBoardGenerator.GenerateRectangeBoard(width, length);
+
+            // Assert
+            string expectedChessBoard = GetExpectedChessBoard(width, length);
+            Assert.Contains(expectedChessBoard, string.Join("\r\n", logger.LoggedMessages));
+        }
+
+        private static string GetExpectedChessBoard(int width, int length)
```

Comment:
Good use. I'd think about moving private methods underneath all the public ones. That's generally the case, with an exception when private method is used once and closely related to public method.